application-analytics-ui/src/main/resources/Analytics/Code/Macros/VelocityMacros.xml

```diff
@@ -267,5 +267,42 @@ API as it is faster
     #end
   #end
 #end
+
+#macro (analytics_createGraph, $id, $options, $class, $date, $apiMethod)
+{{html}}
+&lt;div id="$id" class="container-fluid $class analytics_graphMacro"&gt;
+  &lt;div class="analytics_visitsGenericMacroOptions hidden" data-date="$date" data-method="$apiMethod"&gt;&lt;/div&gt;
```

Comment:
Missing XML escaping. Even if you consider the parameter values as safe, it's always better and safer to escape.